tests/network/firewalld_policy_objects.pm

```diff
@@ -190,7 +188,12 @@ sub run {
     configure_machines($self, $hostname, $net0, $net1);
     barrier_wait 'CONFIGURATION_DONE';
 
+    # Check the Internet connectivity - it may take a moment
+    script_retry('ping -c3 google.com', retry => 12, delay => 5, die => 0);
```

Comment:
You dont need `script_retry`. you still can use `assert_script_run` and pass `-i` and/or `-W` to the `ping` to define some delays and timeouts. This approach doesnt really solve the sporadic issues which the ticket describes. If network is down/broken this `script_retry` is not going to fix it. The positive scenario is tampered and we miss network glitches which they should be reported.

Reply:
No. Sometimes the system needs few attempt to even resolve the domain:
```
# timeout 30 ping -c3 google.com; echo XLTFp-$?-
ping: google.com: Temporary failure in name resolution
XLTFp-2-
# timeout 30 ping -c3 google.com; echo XLTFp-$?-
PING google.com (142.250.186.142) 56(84) bytes of data.
64 bytes from fra24s07-in-f14.1e100.net (142.250.186.142): icmp_seq=1 ttl=111 time=9.30 ms
64 bytes from fra24s07-in-f14.1e100.net (142.250.186.142): icmp_seq=2 ttl=111 time=9.42 ms
64 bytes from fra24s07-in-f14.1e100.net (142.250.186.142): icmp_seq=3 ttl=111 time=9.62 ms

--- google.com ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2004ms
rtt min/avg/max/mdev = 9.302/9.445/9.617/0.130 ms
XLTFp-0-
```
The arguments you mention won't help.

Reply:
Still... but no objection further for now. I am going to approve. Thanks for updating commit message man